lib/services/android-project-service.ts

```diff
@@ -253,7 +255,63 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 	}
 
 	public prepareProject(): IFuture<void> {
-		return Future.fromResult();
+		return (() => {
+			// Delete the AndroidManifest.xml file from res directory as the runtime will consider it as addition to the one in src/main and will try to merge them.
+			// However now they are the same file.
+			let resDestinationDir = this.getAppResourcesDestinationDirectoryPath().wait();
+			let androidManifestPath = path.join(resDestinationDir, this.platformData.configurationFileName);
+			this.$fs.deleteFile(androidManifestPath).wait();
+		}).future<void>()();
+	}
+
+	public ensureConfigurationFileInAppResources(): IFuture<void> {
+		return (() => {
+			let originalAndroidManifestFilePath = path.join(this.$projectData.appResourcesDirectoryPath, this.$devicePlatformsConstants.Android, this.platformData.configurationFileName),
+				hasAndroidManifestInAppResources = this.$fs.exists(originalAndroidManifestFilePath).wait(),
+				shouldExtractDefaultManifest = !hasAndroidManifestInAppResources;
+
+			if(hasAndroidManifestInAppResources) {
+				// Check if the AndroidManifest in app/App_Resouces is the correct one
+				// Use a real magic to detect if this is the correct file, by checking some mandatory strings.
+				let fileContent = this.$fs.readText(originalAndroidManifestFilePath).wait(),
+					isFileCorrect = !!(~fileContent.indexOf("android:minSdkVersion") && ~fileContent.indexOf("android:targetSdkVersion")
+									&& ~fileContent.indexOf("uses-permission") && ~fileContent.indexOf("<application")
+									&& ~fileContent.indexOf("<activity") && ~fileContent.indexOf("<intent-filter>")
+									&& ~fileContent.indexOf("android.intent.action.MAIN") && ~fileContent.indexOf("com.tns.ErrorReportActivity")
+									&& ~fileContent.indexOf("android:versionCode"));
+
+				this.$logger.trace(`Existing ${this.platformData.configurationFileName} is ${isFileCorrect ? "" : "NOT "}correct.`);
+
+				if(!isFileCorrect) {
+					shouldExtractDefaultManifest = true;
```

Comment:
For better readability, we can extract this clock of code into a dedicated function